service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/RemoveSentDocumentService.java

```diff
@@ -0,0 +1,110 @@
+        });
+
+        return asDynamicList(allDocuments, selected, r ->
+            format("%s - %s", r.getPartyName(), format("%s (%s)", r.getDocument().getFilename(), r.getSentAt())));
+    }
+
+
+    public Element<SentDocument> getRemovedSentDocumentById(CaseData caseData, UUID selectedDocId) {
+        return caseData.getDocumentsSentToParties().stream()
+            .map(c -> c.getValue().getDocumentsSentToParty())
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList()).stream().filter(document -> selectedDocId.equals(document.getId()))
+            .findAny()
+            .orElseThrow(() -> new RemovableSentDocumentNotFoundException(selectedDocId));
+    }
+
+    public void populateSentDocumentFields(CaseDetailsMap data, SentDocument sentDocument) {
+        data.put("partyNameToBeRemoved", defaultIfNull(sentDocument.getPartyName(), ""));
+        data.put("sentDocumentToBeRemoved", sentDocument.getDocument());
+        data.put("sentAtToBeRemoved", sentDocument.getSentAt());
+        data.put("letterIdToBeRemoved", sentDocument.getLetterId());
+    }
+
+    public void removeSentDocumentFromCase(CaseData caseData, CaseDetailsMap caseDetailsMap,
+                                           UUID removedDocId) {
+        Element<SentDocument> removedSentDocument = getRemovedSentDocumentById(caseData, removedDocId);
+        final Element<SentDocuments> removedSentDocuments = caseData.getDocumentsSentToParties().stream()
+            .filter(s -> s.getValue().getDocumentsSentToParty().stream().anyMatch(e -> e.equals(removedSentDocument)))
+            .findFirst().orElseThrow();
+
+        caseData.getDocumentsSentToParties().stream().filter(p ->
+                p.getValue().getDocumentsSentToParty().contains(removedSentDocument))
+            .findFirst()
+            .orElseThrow(() -> new RemovableSentDocumentNotFoundException(removedDocId)).getValue()
+            .getDocumentsSentToParty().remove(removedSentDocument);
+        caseDetailsMap.putIfNotEmpty("documentsSentToParties", caseData.getDocumentsSentToParties());
+
+        removedSentDocument.getValue().setRemovalReason(getReasonToRemove(caseData));
+        List<Element<SentDocuments>> hiddenDocumentsSentToParties = caseData.getRemovalToolData()
+                .getHiddenDocumentsSentToParties();
+
+        Optional<Element<SentDocuments>> existingHiddenSentDocuments =
+            ElementUtils.findElement(removedSentDocuments.getId(), hiddenDocumentsSentToParties);
+        if (!existingHiddenSentDocuments.isPresent()) {
+            hiddenDocumentsSentToParties.add(element(removedSentDocuments.getId(), SentDocuments.builder()
+                .partyName(removedSentDocuments.getValue().getPartyName()).build()));
+            existingHiddenSentDocuments =
+                ElementUtils.findElement(removedSentDocuments.getId(), hiddenDocumentsSentToParties);
+        }
+        existingHiddenSentDocuments.ifPresentOrElse(
```

Comment:
because the documents store under the party. Take the following as an example.

```
  "documentsSentToParties": [
    {
      "id": "a1ba061a-2982-4ce3-9881-b74c37aa9b4f",
      "value": {
        "partyName": "Eleanor Graham",
        "documentsSentToParty": [
          {
            "id": "0e133c81-51aa-4bd3-b5aa-7689224ad282",
            "value": {
              "sentAt": "1:26pm, 17 November 2022",
              "document": {
                "document_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/66a3d620-12db-4719-83f7-92f499f098f3",
                "document_filename": "TwoNotice_of_hearing_15November.pdf",
                "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/66a3d620-12db-4719-83f7-92f499f098f3/binary"
              },
              "letterId": "ce601818-6c86-467d-bd4c-ce26186c9272",
              "partyName": "Eleanor Graham",
              "coversheet": {
                "document_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/b48bb5fc-3db8-4024-8ccd-b566da5c5356",
                "document_filename": "Coversheet.pdf",
                "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/b48bb5fc-3db8-4024-8ccd-b566da5c5356/binary"
              }
            }
          }
        ]
      }
    }
  ],
```

```
  "hiddenDocumentsSentToParties": [
    {
      "id": "a1ba061a-2982-4ce3-9881-b74c37aa9b4f",
      "value": {
        "partyName": "Eleanor Graham",
        "documentsSentToParty": [
          {
            "id": "0e133c81-51aa-4bd3-b5aa-7689224ad281",
            "value": {
              "sentAt": "1:26pm, 15 November 2022",
              "document": {
                "document_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/66a3d620-12db-4719-83f7-92f499f098f3",
                "document_filename": "OneNotice_of_hearing_15November.pdf",
                "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/66a3d620-12db-4719-83f7-92f499f098f3/binary"
              },
              "letterId": "ce601818-6c86-467d-bd4c-ce26186c9272",
              "partyName": "Eleanor Graham",
              "coversheet": {
                "document_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/b48bb5fc-3db8-4024-8ccd-b566da5c5356",
                "document_filename": "Coversheet.pdf",
                "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/b48bb5fc-3db8-4024-8ccd-b566da5c5356/binary"
              },
              "removalReason": "Reason"
            }
          }
        ]
      }
    }
  ],
```